packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -404,63 +493,261 @@ const handler: AdaptingProxyHandler<ProxyTarget, EditableTree> = {
+    extends ProxyTarget<FieldAnchor>
+    implements ArrayLike<UnwrappedEditableTree>
+{
+    public readonly fieldKey: FieldKey;
+    public readonly fieldSchema: FieldSchema;
+    public readonly primaryType?: TreeSchemaIdentifier;
+
+    constructor(
+        context: ProxyContext,
+        fieldSchema: FieldSchema,
+        cursor: ITreeSubscriptionCursor,
+        primaryType?: TreeSchemaIdentifier,
+    ) {
+        assert(cursor.mode === CursorLocationType.Fields, "must be in fields mode");
+        super(context, cursor);
+        this.fieldKey = cursor.getFieldKey();
+        this.primaryType = primaryType;
+        this.fieldSchema = fieldSchema;
+    }
+
+    buildAnchorFromCursor(cursor: ITreeSubscriptionCursor): FieldAnchor {
+        return cursor.buildFieldAnchor();
+    }
+
+    tryMoveCursorToAnchor(
+        anchor: FieldAnchor,
+        cursor: ITreeSubscriptionCursor,
+    ): TreeNavigationResult {
+        return this.context.forest.tryMoveCursorToField(anchor, cursor);
+    }
+
+    forgetAnchor(anchor: FieldAnchor): void {
+        if (anchor.parent === undefined) return;
+        this.context.forest.anchors.forget(anchor.parent);
+    }
+
+    readonly [index: number]: UnwrappedEditableTree;
+
+    public get length(): number {
+        return this.cursor.getFieldLength();
+    }
+
+    /**
+     * Returns a node (unwrapped by default, see {@link UnwrappedEditableTree}) by its index.
+     */
+    public proxifyNode(index: number, unwrap = true): UnwrappedEditableTree {
+        this.cursor.enterNode(index);
+        const target = new NodeProxyTarget(this.context, this.cursor);
+        this.cursor.exitNode();
+        return inProxyOrUnwrap(target, unwrap);
+    }
+
+    /**
+     * Gets a node by its index without unwrapping.
+     */
+    public getWithoutUnwrapping(index: number): EditableTree {
+        assert(
+            keyIsValidIndex(index, this.length),
+            "A child node must exist at index to get it without unwrapping.",
+        );
+        return this.proxifyNode(index, false) as EditableTree;
+    }
+
+    /**
+     * Gets array of unwrapped nodes.
+     */
+    private get asArray(): UnwrappedEditableTree[] {
+        return mapCursorField(this.cursor, (c) =>
+            inProxyOrUnwrap(new NodeProxyTarget(this.context, c), true),
+        );
+    }
+
+    *[Symbol.iterator](): IterableIterator<UnwrappedEditableTree> {
+        for (const node of this.asArray) {
+            yield node;
+        }
+    }
+}
+
+/**
+ * Returns a Proxy handler, which together with a {@link FieldProxyTarget} implements a basic read/write access to
+ * the sequence fields by means of the cursors.
+ */
+const fieldProxyHandler: AdaptingProxyHandler<FieldProxyTarget, EditableField> = {
+    get: (target: FieldProxyTarget, key: string | symbol, receiver: object): unknown => {
+        if (typeof key === "string") {
+            if (key in { length: true, fieldKey: true, fieldSchema: true, primaryType: true }) {
```

Comment:
it would be better to use a Set rather than a JS object for this. "in" will return true for keys not listed here, such a "__proto__" or "toString". If this is intended behavior, it should be documented since its not obvious.

Reply:
Done.